src/gsy_e/models/strategy/energy_parameters/heat_pump_tank.py

```diff
@@ -0,0 +1,311 @@
+    ) -> TankSolverParameters:
+        """
+        Return tank solver parameters, that can be used by the solver to maintain the current tank
+        temp.
+        """
+        current_storage_temp_C = self._state.get_storage_temp_C(time_slot)
+        target_storage_temp_C = current_storage_temp_C
+        if not self._parameters.min_temp_C < target_storage_temp_C < self._parameters.max_temp_C:
+            logger.info(
+                "Storage temp %s cannot exceed min (%s) / max (%s) tank temperatures.",
+                target_storage_temp_C,
+                self._parameters.min_temp_C,
+                self._parameters.max_temp_C,
+            )
+            target_storage_temp_C = max(
+                min(target_storage_temp_C, self._parameters.max_temp_C),
+                self._parameters.min_temp_C,
+            )
+        return TankSolverParameters(
+            tank_volume_L=self._parameters.tank_volume_L,
+            current_storage_temp_C=current_storage_temp_C,
+            target_storage_temp_C=target_storage_temp_C,
+        )
+
+    def create_tank_parameters_for_maxing_tank_temp(
+        self, time_slot: DateTime
+    ) -> TankSolverParameters:
+        """
+        Return tank solver parameters, that can be used by the solver to increase the current tank
+        temp to its maximum.
+        """
+        current_storage_temp_C = self._state.get_storage_temp_C(time_slot)
+        target_storage_temp_C = self._parameters.max_temp_C
+        return TankSolverParameters(
+            tank_volume_L=self._parameters.tank_volume_L,
+            current_storage_temp_C=current_storage_temp_C,
+            target_storage_temp_C=target_storage_temp_C,
+        )
+
+    def create_tank_parameters_without_target_tank_temp(self, time_slot: DateTime):
+        """
+        Create tank parameters without specifying tank temp, in order to be used as an output of
+        the solver.
+        """
+        current_storage_temp_C = self._state.get_storage_temp_C(time_slot)
+        return TankSolverParameters(
+            tank_volume_L=self._parameters.tank_volume_L,
+            current_storage_temp_C=current_storage_temp_C,
+        )
+
+    def _temp_diff_to_Q_kWh(self, diff_temp_K: float) -> float:
+        return diff_temp_K * self._Q_specific
+
+    def _Q_kWh_to_temp_diff(self, energy_kWh: float) -> float:
+        return energy_kWh / self._Q_specific
+
+
+class AllTanksEnergyParameters:
+    """Manage the operation of heating and extracting temperature from multiple tanks."""
+
+    def __init__(self, tank_parameters: List[TankParameters]):
+        self._tanks_energy_parameters = [
+            TankEnergyParameters(tank, GlobalConfig.slot_length) for tank in tank_parameters
+        ]
+
+    def increase_tanks_temp_from_heat_energy(self, heat_energy: float, time_slot: DateTime):
+        """Increase the temperature of the water tanks with the provided heat energy."""
+        # Split heat energy equally across tanks
```

Comment:
Nothing to do now, but we might implement a mechanism to distribute the produced heat differently for different tanks, e.g. with coefficients for each tank (the sum over all coefficients is 1).
But only if this is a requirement of course.

Reply:
I was also thinking about that, and I also thought about equally simplistic alternatives, e.g. distribute heat proportionally to the tank size. However, heat share across tanks depends largely on the heat exchanger and plumbing / connections between the heatpump and each water tank. Since we omitted simulating even the heat exchanger though, I decided to skip it altogether and share the heat proportionally across tanks. 
If it becomes a requirement though you are right, this definitely qualifies a new mechanism in a separate class that decides on the distribution of heat amongst the water tanks.